src/main.rs

```diff
@@ -66,15 +66,17 @@ enum Cmd {
     #[structopt(visible_alias = "e")]
     Encode {
         /// An id or the name of a type that exists in the type registry
-        #[structopt(short, required_unless = "pallet-call")]
-        ty: Option<String>,
+        #[structopt(short, required_unless = "pallet-call", default_value = "0")]
+        ty: String,
         /// Indicates that the input values are the parameters of `<pallet>/<call>`
         #[structopt(short = "c", conflicts_with = "ty")]
         pallet_call: Option<String>,
```

Comment:
It should. But somehow, structopt doesn's support `required_unless` for multiple keywords, less for subcommands. So, in this case, if you put `required_unless = "pallet-call"` and at the same time `required_unless = "cmd"`, would always ignore the latter, regardless of the order, and always make `ty` to be a required option.

Also, if you were to specify the option keyword (like `extrinsic`), the macro would fail at compile time.

This is related with [this](https://github.com/virto-network/sube/pull/18#discussion_r1051209641). The default value was the best way to circumvent the lack of optionality of this value, and at the same time, the necessity of it.

That said, it's ambiguous, and maybe we'd need to rethink the structure of that command to resolve it. Maybe work with a `decode extrinsic`, and then, a `decode type` subcommand that requires the type as a required argument for it.

Reply:
Done this way!